components/favorite-button.tsx

```
@@ -0,0 +1,162 @@
+				'group relative flex items-center justify-center rounded-full transition-all duration-300 ease-out',
+				'bg-white/80 dark:bg-gray-900/80 backdrop-blur-sm border border-gray-200/50 dark:border-gray-700/50',
+				'hover:bg-white dark:hover:bg-gray-800 hover:border-gray-300 dark:hover:border-gray-600',
+				'shadow-sm hover:shadow-md active:scale-95',
+				'focus:outline-none focus:ring-2 focus:ring-blue-500/50 focus:ring-offset-2 dark:focus:ring-offset-gray-900',
+				'disabled:opacity-50 disabled:cursor-not-allowed',
+				variant === 'star' && sizeClasses[size],
+				variant === 'star' && positionClasses[position],
+				className
+			)}
+			aria-label={isFav ? `Remove ${itemName} from favorites` : `Add ${itemName} to favorites`}
+		>
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)